src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -189,7 +191,7 @@ private DiagnosticInfo ToFileReadDiagnostics(Exception e, CommandLineSourceFile
             return diagnosticInfo;
         }
 
-        internal bool PrintErrors(IEnumerable<Diagnostic> diagnostics, TextWriter consoleOutput)
+        protected bool ReportErrors(IEnumerable<Diagnostic> diagnostics, TextWriter consoleOutput, ErrorLogger errorLogger)
```

Comment:
how about rather than ReportErrors know what to report per specific logger type, ReportErrors (IEnumerable<Diagnostics>, IEnumerable<Logger> loggers) or you create aggregatedLogger and then

foreach(var logger in loggers)
{ logger.Log(diag, this.Culture); } 

or if you chose aggregatedLogger pattern, just do logger.Log(diag, this.Culture) ?